aws/src/database/multi_table/update_cautionary_alerts.py

```diff
@@ -0,0 +1,60 @@
+
+from mypy_boto3_dynamodb.service_resource import Table
+
+from aws.src.database.dynamodb.utils.get_dynamodb_table import get_dynamodb_table
+from aws.src.utils.csv_to_dict_list import csv_to_dict_list
+from enums.enums import Stage
+from dateutil import parser
+from datetime import datetime
+
+STAGE = Stage.HOUSING_PRODUCTION
+
+def update_cautionary_alert_in_db_if_tenure_inactive(person_table: Table, alerts_from_csv: list[dict]):
+    """
+    Iterate through the alerts and remove property details if tenure is inactive
+    1. Get person details from dynamo using mmh_id in csv
+    2. check if there is an end date within the tenure object
+    3. if so remove the property details from the csv
+    """
+
+    for alert_item in alerts_from_csv:
+        alert_id = alert_item["id"]
+        mmh_id = alert_item["mmh_id"]
+        if mmh_id is '' or None:
+            alert_item["failed_reason"] = f"mmh_id is null for id {alert_id}"
+            continue
+        person = person_table.get_item(Key={"id": mmh_id}).get("Item")
+
+        person_id = person["id"]
+        if person is None:
```

Comment:
Could be wrong, but...
If the person is indeed `None`, then the `line 29` will never be reached due to a **runtime error** at `line 28`. That's because `line 28` is trying to do: `None["id"]` - a C# "Null Reference Exception" equivalent.

Reply:
I dont think this is true, as when I tried it would it did give the exception. Also, this is highly unexpected as the person would have an object in all scenarios

Reply:
If
>"person would have an object in all scenarios"

then why check?

``` py
if person is None:
```

You'd only do this check if person can sometimes be None, and if it can, then line 28 should be below the 29.

Reply:
I suppose, my point is... the Line 29 does nothing at all:
1. If person can be None, then this line will never be reached, because script would fail on line 28.
2. If person cannot be None, then this line is not needed, because: "why would you check something that's already certain to be false?"

Reply:
It was added there as a just incase scenario, I guess I see your point so I'll just remove that part